### PR TITLE
[5.6] Add quantile method and calculate the median based on it

### PR DESCRIPTION
**Pull request by krenor**
The median is the same as the 0.5 quantile of all values.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -139,31 +139,44 @@ public function average($callback = null)
...
      *
-     * @param  null $key
-     * @return mixed
+     * @param  float $quantile
+     * @param  string|array|null $key
+     *
+     * @return int|float|void
      */
-    public function median($key = null)
+    public function quantile(float $quantile, $key = null)
```
Comment (afraca) [inline src/Illuminate/Support/Collection.php]: The framework does no (scalar) typehinting, so you can leave off the `float`

File: src/Illuminate/Support/Collection.php [hunk 2]
```diff
@@ -141,12 +141,12 @@ public function average($callback = null)
     /**
      * Calculate the quantile of a given key.
      *
-     * @param  float $quantile
+     * @param  float|int  $quantile
```
Comment (krenor) [inline src/Illuminate/Support/Collection.php]: This cannot be an `int` as its required to be between 0 and 1

File: src/Illuminate/Support/Collection.php (hunk 1 above)
Comment (krenor) [reply src/Illuminate/Support/Collection.php]: Yeah, I'm to used throwing scalar hints everywhere :sweat_smile:

File: src/Illuminate/Support/Collection.php (hunk 2 above)
Comment (krenor) [reply src/Illuminate/Support/Collection.php]: Yeah, you're right as it'll just result in null. :)

**Comment by taylorotwell** [comment]
No plans on adding this to core.

**Pull request closed without merging**